CONTRIBUTING.md

```diff
@@ -37,14 +33,47 @@ these actions are configured to run on your fork instead of on the base reposito
 
 3. Make sure the branch of your fork where you push your changes is not called `master`. If it is, either change its name or remember to manually trigger the `Code & Documentation Updates` action after a push.
 
-### Local
-However, you can also run the tests locally by executing pytest in your terminal from the `/test` folder.
+## Setting up your development environment
+
+When working on Haystack, we recommend installing it in editable mode with `pip install -e` in a Python virtual
+environment. From the root folder:
+```
+pip install -e '.[test]'
+```
+
+This will install all the dependencies you need to work on the codebase, which most of the times is a subset of all the
+dependencies needed to run Haystack.
+
+## Running the tests
+
+Tests will automatically run in our CI for every commit you push to your PR on Github. This is usually the most convenient
+way and we encourage you to create early "draft pull requests" to leverage the CI at an early stage.
+
+Tests can also be executed locally, by launching `pytest` from the `/test` folder (this is important because running from the
+root folder would also execute the ui and rest API tests, that require specific dependencies).
+
+You can control which tests to run using Pytest markers, let's see how.
+
+### Running a subset of tests (recommended)
+
+In most cases you rather want to run a **subset of tests** locally that are related to your dev, and the most important
+option to reduce the number of tests in a meaningful way, is to run tests only for a list of **selected document stores**.
+This is possible by adding the `--document_store_type` arg to your `pytest` command (possible values are: `"elasticsearch, faiss, memory, milvus, weaviate"`).
 
-#### Running all tests
-**Important**: If you want to run **all** tests locally, you'll need **all** document stores running in the background before you run the tests.
-Many of the tests will then be executed multiple times with different document stores.
+For example, calling `pytest . --document_store_type="memory"` will run all the document store tests using the
+InMemoryDocumentStore only, skipping the others (the logs will show which ones). The `InMemoryDocument` store is a very
+good starting point as it doesn't require any external resource:
+```
+pytest . --document_store_type="memory"
+```
 
-You can launch them like this:
+You can also run the tests using a **combination of document stores**, provided the corresponding services are up and
```

Comment:
Having docker running should be enough. Our `launch_es()` should start an elasticsearch docker container.